addon/components/attach-popover.js

```diff
@@ -1,18 +1,166 @@
 import Component from '@ember/component';
-import layout from '../templates/components/ember-attacher-inner';
+import DEFAULTS from '../defaults';
+import layout from '../templates/components/attach-popover';
 import { cancel, debounce, later, next, run } from '@ember/runloop';
 import { computed, observer } from '@ember/object';
+import { getOwner } from '@ember/application';
+import { guidFor } from '@ember/object/internals';
 import { htmlSafe } from '@ember/string';
+import { stripInProduction } from 'ember-attacher/-debug/helpers';
+import { warn } from '@ember/debug';
 
 export default Component.extend({
+  layout,
+
+  tagName: '',
+
   /**
    * ================== PUBLIC CONFIG OPTIONS ==================
    */
 
-  // See ember-attacher.js, which passes all the default values into this component
+  animation: DEFAULTS.animation,
+  arrow: computed('animation', {
+    get() {
+      return DEFAULTS.arrow;
+    },
+
+    set(_, val) {
+      stripInProduction(() => {
+        if (this.get('animation') === 'fill' && val) {
+          warn('Animation: \'fill\' is not compatible with arrow: true', { id: 70015 });
+        }
+      });
+
+      return val;
+    }
+  }),
+  class: DEFAULTS.class,
+  hideDelay: DEFAULTS.hideDelay,
+  hideDuration: DEFAULTS.hideDuration,
+  hideOn: DEFAULTS.hideOn,
+  interactive: DEFAULTS.interactive,
+  isOffset: DEFAULTS.isOffset,
+  isShown: DEFAULTS.isShown,
+  lazyRender: DEFAULTS.lazyRender,
+  onChange: null,
+  placement: DEFAULTS.placement,
+  popperContainer: DEFAULTS.popperContainer,
+  popperOptions: DEFAULTS.popperOptions,
+  renderInPlace: DEFAULTS.renderInPlace,
+  showDelay: DEFAULTS.showDelay,
+  showDuration: DEFAULTS.showDuration,
+  showOn: DEFAULTS.showOn,
+  target: null,
 
   /**
-   * ================== COMPONENT LIFECYCLE HOOKS ==================
+   * ================== PRIVATE IMPLEMENTATION DETAILS ==================
+   */
+
+  actions: {
+    // Exposed via the named yield to enable custom hide events
+    hide() {
+      this._hide();
+    },
+
+    registerAPI(api) {
```

Comment:
I should make everything in here a normal property, except for `target`, since there are observers that depend on it